floodviz/linked_data_utils.py

```diff
@@ -0,0 +1,150 @@
+        return l
+
+    def _assemble_event(self):
+        """
+        Wrap the data on the event into a dictionary
+        :return: JSON-LD-like dict representing the event
+        """
+        event = self._blank_thing("Event")
+        event.update({
+            "@context": "http://schema.org",
+            "@type": "Event",
+            "name": "FLOOD EVENT NAME",
+            "startDate": self.dates['start'],
+            "endDate": self.dates['end'],
+            "location": {
+                "@context": "http://schema.org",
+                "@type": "Place",
+                "address": "null",
+                "geo": {
+                    "@context": "http://schema.org",
+                    "@type": "GeoShape",
+                    "box": self._location_str(),
+                },
+            },
+        })
+        return event
+
+    def _assemble_gage(self, gage):
+        """
+        Wrap an individual gage in a place
+        :param gage: the gage to be wrapped
+        :return: A dict representing the gage in json-ld format as a place
+        """
+        g = self._blank_thing('Place')
+        geo = self._blank_thing('geoCoordinates')
+        geo.update({
+            "longitude": gage['dec_long_va'],
+            "latitude": gage['dec_lat_va']
+        })
+        g.update({
+            "address": "HUC:" + gage['huc_cd'],
+            "name": gage['station_nm'],
+            "branchCode": "SITE:"+gage['site_no'],
+            "geo": geo,
+            "additionalProperty": {
+                "huc_cd": gage['huc_cd'],
+                "site_no": gage['site_no']
+            }
+        })
+        return g
+
+    def _assemble_all_gages(self):
+        """
+        Wrap up all the gages as places
+        :return: A list of dicts describing the gages
+        """
+        gages_ld = []
+        for gage in self.gages:
+            gages_ld.append(self._assemble_gage(gage))
+        return gages_ld
+
+    def set_gages(self, gages):
+        """
+        Sets the gages to be used
+        :param gages: list of dicts describing gages as output by `site_dict` in map_utils.
+        :return: None
+        """
+        self.gages = gages
+
+    def set_dates(self, start, end):
+        """
+        Sets the start and end dates of the flood event
+        :param start: Start date
+        :param end: End date
+        :return: None
+        """
+        self.dates = {
+            "start": start,
+            "end": end
+        }
+
+    def set_location(self, bbox):
+        """
+        Sets the bounding box of the event
+        :param bbox: array containing two pairs of coordinates
+        :return: None
+        """
+        lon = [bbox[0], bbox[2]]
+        lat = [bbox[1], bbox[3]]
+        minlat = min(lat)
+        maxlat = max(lat)
+        minlon = min(lon)
+        maxlon = max(lon)
+
+        self.location = [minlat, minlon, maxlat, maxlon]
+
+    def assemble(self):
+        """
```

Comment:
maybe have a check here to see if things like location and gauges have been set yet? Just thinking in the context of them reusing this class in a separate project. Or you could do this in the private methods you call here.